vignettes/satterthwaite.Rmd

```diff
@@ -0,0 +1,122 @@
+`mmrm`](algorithms.html).
+
+We are also using the same notation as in the [Details of the
+Kenward-Roger calculations](kenward.html). In particular, we assume we
+have a contrast matrix $C \in \mathbb{R}^{c\times p}$ with which we want
+to test the linear hypothesis $C\beta = 0$. Further, $W(\hat\theta)$ is
+the inverse of the Hessian matrix of the log-likelihood function of
+$\theta$ evaluated at the estimate $\hat\theta$, i.e. the observed
+Fisher Information matrix as a consistent estimator of the
+variance-covariance matrix of $\hat\theta$.
+$\Phi(\theta) = \left\{X^\top \Omega(\theta)^{-1} X\right\} ^{-1}$ is
+the asymptotic covariance matrix of $\hat\beta$.
+
+## One-dimensional contrast
+
+We start with the case of a one-dimensional contrast, i.e. $c = 1$. The
+Satterthwaite adjusted degrees of freedom for the corresponding t-test
+are then defined as:
+\[
+\hat\nu(\hat\theta) = \frac{2f(\hat\theta)^2}{f{'}(\hat\theta)^\top W(\hat\theta) f{'}(\hat\theta)}
+\]
+where $f(\hat\theta) = C \Phi(\hat\theta) C^\top$ is the scalar in the
+numerator and we can identify it as the variance estimate for the
+estimated scalar contrast $C\hat\beta$. The computational challenge is
+essentially to evaluate the denominator in the expression for
+$\hat\nu(\hat\theta)$, which amounts to computing the $k$-dimensional
+gradient $f{'}(\hat\theta)$ of $f(\theta)$ (for the given contrast
+matrix $C$) at the estimate $\hat\theta$. We already have the
+variance-covariance matrix $W(\hat\theta)$ of the variance parameter
+vector $\theta$ from the model fitting.
+
+### Jacobian approach
+
+However, if we proceeded in a naive way here, we would need to recompute
+the denominator again for every chosen $C$. This would be slow, e.g.
+when changing $C$ every time we want to test a single coefficient within
+$\beta$. It is better to instead evaluate the gradient of the matrix
+valued function $\Phi(\theta)$, which is therefore the Jacobian, with
+regards to $\theta$, $\mathcal{J}(\theta) = \nabla_\theta \Phi(\theta)$.
+Imagine $\mathcal{J}(\theta)$ as the the 3-dimensional array with $k$
+faces of size $p\times p$. Left and right multiplying each face by $C$
+and $C^\top$ respectively leads to the $k$-dimensional gradient
+$f'(\theta) = C \mathcal{J}(\theta) C^\top$. Therefore for each
+new contrast $C$ we just need to perform simple matrix multiplications,
+which is fast (see `h_gradient()` where this is implemented). Thus,
+having computed the estimated Jacobian $\mathcal{J}(\hat\theta)$,
+it is only a matter of putting the different quantities together to
+compute the estimate of the denominator degrees of freedom,
+$\hat\nu(\hat\theta)$.
+
+### Jacobian calculation
+
+Currently, we evaluate the gradient of $\Phi(\theta)$ (which is created
+as an R function using `h_covbeta_fun()`) with regards to $\theta$
+evaluated at $\hat\theta$, i.e. the Jacobian $\mathcal{J}(\hat\theta)$,
+numerically using the `jacobian` function from the `numDeriv` package
```

Comment:
when we need the `()` in referencing the functions, when we don't need?I am a bit confused

Reply:
hm good question :-) In the roxygen documentation we naturally use it with parentheses for functions, because that is how the md syntax works. So I would suggest to do the same in the vignettes. Let me add the hyperlinks here as well though.

Reply:
hm I can't because these are not exported and therefore not part of pkgdown reference. ok

Reply:
oh I see... these [function()] only work for internal function references in the package, and for external function a simple `function` is sufficient. Thank you!

Reply:
well it is complicated - in Roxygen, we can use `[function()]` also for internal functions, because we have documentation for all of them. However in the vignette, we cannot reference help pages but only html populated on the pkgdown website. but that includes only exported functions. but to make it look similar to the others I would say here also use `function()` style. I am adding a paragraph in the contributing guide for this now